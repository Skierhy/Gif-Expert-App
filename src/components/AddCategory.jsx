import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
	const [inputValue, setInputValue] = useState('');

	const onInputChange = (event) => {
		setInputValue(event.target.value);
	};

	const onSubmit = (event) => {
		event.preventDefault();
		if (inputValue.trim().length <= 0) {
			return;
		}
		// setCategories((categories) => {
		// 	return [inputValue, ...categories];
		// });
		onNewCategory(inputValue);
		setInputValue('');
	};

	return (
		<form onSubmit={(event) => onSubmit(event)} aria-label="form">
			<input
				type="text"
				placeholder="Buscar Gifs"
				value={inputValue}
				onChange={(event) => onInputChange(event)}
			/>
		</form>
	);
};

AddCategory.propTypes = {
	onNewCategory: PropTypes.func.isRequired,
};
