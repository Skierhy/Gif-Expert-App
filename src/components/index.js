// lo que hace es exporta todo lo que exporta el archivo
export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';
