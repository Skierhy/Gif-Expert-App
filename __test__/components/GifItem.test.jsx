import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas de GifItem', () => {
	const title = 'Holaa';
	const url = 'https://';
	test('Prueba de snapshots', () => {
		const { container } = render(<GifItem title={title} url={url} />);
		expect(container).toMatchSnapshot();
	});

	test('Debe que mostrar la imagen con el url y el alt indicado', () => {
		render(<GifItem title={title} url={url} />);
		// screen.debug();
		const { alt, src } = screen.getByRole('img');
		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('Debe que mostrar el titulo mostrado en un pÃ¡rrafo', () => {
		render(<GifItem title={title} url={url} />);
		expect(screen.getByText(title)).toBeTruthy();
		expect(screen.getByText(title).innerHTML).toBe(title);
	});
});
