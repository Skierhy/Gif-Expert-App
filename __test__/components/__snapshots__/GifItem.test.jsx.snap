// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifItem Prueba de snapshots 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Holaa"
      src="https://"
    />
    <p>
      Holaa
    </p>
  </div>
</div>
`;
